sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -1328,6 +1329,41 @@ def build_spec_by_group(
     )
 
 
+def modify_task_for_ignore_upstream_failure(
+    task: pipeline_task.PipelineTask,
+    pipeline_task_spec: pipeline_spec_pb2.PipelineTaskSpec,
+):
+    if task._ignore_upstream_failure_tag:
+        pipeline_task_spec.trigger_policy.strategy = (
+            pipeline_spec_pb2.PipelineTaskSpec.TriggerPolicy.TriggerStrategy
+            .ALL_UPSTREAM_TASKS_COMPLETED)
+
+        for input_name, input_spec in (task.component_spec.inputs or
+                                       {}).items():
+            if type_utils.is_task_final_status_type(input_spec.type):
```

Comment:
nit: consider reducing the level of nesting via
```python
if not ...
     continue
```

Reply:
Thanks! Done.